nvim/types.go

```diff
@@ -0,0 +1,245 @@
+// Copyright 2019 Gary Burd
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package nvim
+
+// Mode represents a Nvim's current mode.
+type Mode struct {
+	// Mode is the current mode.
+	Mode string `msgpack:"mode"`
+
+	// Blocking is true if Nvim is waiting for input.
+	Blocking bool `msgpack:"blocking"`
+}
```

Comment:
why make a new type for this? This seems over-specific. Can't it just be a map? Or would that require `interface{}` ?

Reply:
https://github.com/neovim/go-client/pull/36#discussion_r278015661 leads me to believe this will be forward-compatible. If so, OK.

Reply:
@justinmk I didn't add a new type.
Just move some types to `types.go` such as https://github.com/neovim/go-client/blob/fba3ac82e748917b9367dea01aad6525ac5e300e/nvim/nvim.go#L645-L691